test/end2end_test.go

```diff
@@ -1484,6 +1484,38 @@ func testMetadataUnaryRPC(t *testing.T, e env) {
 	}
 }
 
+func TestPeerClientSide(t *testing.T) {
```

Comment:
Move this to above TestMetadataUnaryRPC. So that all metadata related tests stay together.